Bell State Stabilization with Real-time Feedback/Bell_State_Stabilization_w_Realtime_Feedback.ipynb

```diff
@@ -0,0 +1,1422 @@
+    "\n",
+    "\n",
+    "Copyright (C) 2021 Zurich Instruments\n",
+    "This software may be modified and distributed under the terms of the MIT license. See the LICENSE file for details."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Table of Contents:\n",
+    "* [1. Experimental Setup](#setup)\n",
+    "* [2. Import Packages](#import)\n",
+    "* [3. Initialization](#initialization)\n",
+    "* [4. Readout Parameters](#readout)\n",
+    "* [5. PQSC Configuration](#pqsc_config)\n",
+    "* [6. HDAWG Configuration](#hdawg_config)\n",
+    "    * [6.1 Configuration of HDAWG 1 AWG 0 for Control of Data Qubit D<sub>1</sub>](#hdawg1_awg0_config)\n",
+    "    * [6.2 Configuration of HDAWG 2 AWG 0 for Control of Data Qubit D<sub>2</sub>](#hdawg2_awg0_config)\n",
+    "    * [6.3 Configuration of HDAWG 1 AWG 1 for Control of Z-Ancilla](#hdawg1_awg1_config)\n",
+    "    * [6.4 Configuration of HDAWG 2 AWG 1 for Control of X-Ancilla](#hdawg2_awg1_config)\n",
+    "* [7. UHFQA Configuration](#uhfqa_config)\n",
+    "* [8. Main Program](#main)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Experimental Setup: <a class=\"anchor\" id=\"setup\"></a>\n",
+    "Here is the required experimental setup, including all the wiring.\n",
+    "\n",
+    "![The setup](experimental_setup.svg)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The assignments of the devices for certain tasks are as follows:\n",
+    "\n",
+    "![Bell state stabilization](channel_assignments.svg)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Import Packages <a class=\"anchor\" id=\"import\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Import zhinst-qcodes drivers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import zhinst.qcodes as ziqc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Import other external packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import textwrap\n",
+    "import itertools"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Initialization <a class=\"anchor\" id=\"initialization\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* Set the data server host address\n",
+    "* The serial numbers of the instruments to run the script with"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Dataserver host\n",
+    "dataserver_host = \"localhost\"\n",
+    "\n",
+    "# Instrument serials\n",
+    "serial_hdawg_1 = \"dev8030\"\n",
```

Comment:
please replace the device id by generic values, so dev8001 for hdawg_1,  dev8002 for hdawg_2,  dev2001 for uhfqa and  dev10001 for pqsc